kimchi/src/folding/mod.rs

```diff
@@ -161,18 +179,45 @@ impl<'a, F: Clone> EvalLeaf<'a, F> {
     }
 }
 
+/// Describe a folding environment.
+/// The type parameters are:
+/// - `F`: The field of the circuit/computation
+/// - `I`: The instance type, i.e the public inputs
+/// - `W`: The type of the witness, i.e. the private inputs
+/// - `Col`: The type of the column
+/// - `Chal`: The type of the challenge
 pub trait FoldingEnv<F, I, W, Col, Chal> {
+    /// Structure which could be storing useful information like selectors, etc.
     type Structure;
-    /// A vec of just zeros of the same length as other columns
+
+    /// Creates a new environment storing the structure, instances and witnesses.
+    fn new(structure: &Self::Structure, instances: [&I; 2], witnesses: [&W; 2]) -> Self;
+
+    // TODO: move into `FoldingConfig`
+    // FIXME: when we move this to `FoldingConfig` it will be general for all impls as:
+    // vec![F::zero(); Self::rows()]
+    /// Returns a vector of zeros with the same length as the number of rows in the circuit.
     fn zero_vec(&self) -> Vec<F>;
+
+    /// Returns the evaluations of a given column witness at omega or zeta*omega.
     fn col(&self, col: Col, curr_or_next: CurrOrNext, side: Side) -> &Vec<F>;
+
+    // TODO: could be shared across circuits of the same type
+    /// Returns the evaluations of the i-th lagrangian term.
```

Comment:
```suggestion
    /// Returns the evaluations of the i-th Lagrangian term.
```
Neat: I suppose we should use capital `L`?

Reply:
Even maybe simply `Lagrange`? Not sure we can make it an adjective from the name :).

Reply:
Looks like capital letter is the way to go, but Lagrangian does exist indeed -> https://en.wikipedia.org/wiki/Lagrangian